src/plugins/intel_npu/src/al/include/intel_npu/al/icompiled_model.hpp

```diff
@@ -17,16 +17,14 @@ class ICompiledModel : public ov::ICompiledModel {
 public:
     using ov::ICompiledModel::ICompiledModel;
 
-    virtual const std::shared_ptr<const NetworkDescription>& get_network_description() const = 0;
+    virtual const std::vector<uint8_t>& get_compiled_network() const = 0;
```

Comment:
To support new IGraph structure, have to change all Interface class to use the new class instead of NetworkDescription, but they are semantically equivalent, the new change may break npu build validation since it is not compatible.

Have to change other backends if we do this.